server/src/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1104,14 +1107,61 @@ public void cleanupStorage(boolean recurring) {
                     }
 
                     // remove snapshots in Error state
-                    List<SnapshotVO> snapshots = _snapshotDao.listAllByStatus(Snapshot.State.Error);
+                    List<SnapshotVO> snapshots = _snapshotDao.listAllByStatusIncludingRemoved(Snapshot.State.Error);
                     for (SnapshotVO snapshotVO : snapshots) {
                         try {
                             List<SnapshotDataStoreVO> storeRefs = _snapshotStoreDao.findBySnapshotId(snapshotVO.getId());
+                            boolean isVMware = snapshotVO.getHypervisorType().equals(HypervisorType.VMware);
```

Comment:
please try to move this logic into the vmware plugin. this code should remain generic.
opinions @mike-tutkowski @swill @rhtyd ?